src/Containers/Microsoft.NET.Build.Containers/ContainerBuilder.cs

```diff
@@ -70,7 +74,31 @@ public static class ContainerBuilder
         Layer newLayer = Layer.FromDirectory(publishDirectory.FullName, workingDir, imageBuilder.IsWindows);
         imageBuilder.AddLayer(newLayer);
         imageBuilder.SetWorkingDirectory(workingDir);
-        imageBuilder.SetEntrypointAndCmd(entrypoint, cmd ?? Array.Empty<string>());
+
+        bool hasErrors = false;
+        (string[] imageEntrypoint, string[] imageCmd) = ImageBuilder.DetermineEntrypointAndCmd(entrypoint, entrypointArgs, defaultArgs, appCommand, appCommandArgs, appCommandInstruction,
```

Comment:
is it more reasonable to have this validation in parser itself?

Reply:
The reason I placed it here is because the base image entrypoint is passed as an argument to this method.